test/others/libcriu should not use criu internal headers

```
2026-03-03T15:10:05.8524426Z + make -C test/others/libcriu run
2026-03-03T15:10:05.8525002Z make: Entering directory '/criu/test/others/libcriu'
2026-03-03T15:10:05.8525646Z gcc -c test_sub.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_sub.o -Werror
2026-03-03T15:10:05.8909826Z gcc -c lib.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o lib.o -Werror
2026-03-03T15:10:05.9219947Z gcc test_sub.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_sub
2026-03-03T15:10:05.9420819Z gcc -c test_self.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_self.o -Werror
2026-03-03T15:10:05.9795095Z gcc test_self.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_self
2026-03-03T15:10:05.9995319Z gcc -c test_notify.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_notify.o -Werror
2026-03-03T15:10:06.0376293Z gcc test_notify.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_notify
2026-03-03T15:10:06.0576390Z gcc -c test_rpc_config.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_rpc_config.o -Werror
2026-03-03T15:10:06.1033096Z gcc test_rpc_config.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_rpc_config
2026-03-03T15:10:06.1235871Z gcc -c test_iters.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_iters.o -Werror
2026-03-03T15:10:06.1650169Z gcc test_iters.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_iters
2026-03-03T15:10:06.1850502Z gcc -c test_errno.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_errno.o -Werror
2026-03-03T15:10:06.2263047Z gcc test_errno.o lib.o -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o test_errno
2026-03-03T15:10:06.2465930Z gcc -c test_join_ns.c -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o test_join_ns.o -Werror
2026-03-03T15:10:06.2590721Z In file included from /usr/include/bits/fcntl-linux.h:468,
2026-03-03T15:10:06.2591289Z                  from /usr/include/bits/fcntl.h:61,
2026-03-03T15:10:06.2591741Z                  from /usr/include/fcntl.h:35,
2026-03-03T15:10:06.2592139Z                  from test_join_ns.c:3:
2026-03-03T15:10:06.2592841Z ../../../../criu/criu/include/linux/openat2.h:6:10: fatal error: common/config.h: No such file or directory
2026-03-03T15:10:06.2593572Z     6 | #include "common/config.h"
2026-03-03T15:10:06.2593914Z       |          ^~~~~~~~~~~~~~~~~
2026-03-03T15:10:06.2594258Z compilation terminated.
```

--- a/test/others/libcriu/Makefile
+++ b/test/others/libcriu/Makefile
@@ -11,6 +11,16 @@ TESTS += test_pre_dump
 TESTS += test_check
 TESTS += test_feature_check
 
+# Install libcriu (headers and library) into a local staging directory so
+# that nothing from the non-test part of the repo is referenced directly
+# during the build or at runtime.
+INSTALL_DIR:= $(CURDIR)/.install
+CRIU_INCLUDE:= $(INSTALL_DIR)/include/criu
+CRIU_LIB:= $(INSTALL_DIR)/lib
+
+CFLAGS:= -I$(CRIU_INCLUDE)
+LDFLAGS:= -L$(CRIU_LIB) -lcriu
+
 all: $(TESTS)
 .PHONY: all
 
@@ -18,24 +28,24 @@ run: all
 	./run.sh
 .PHONY: run
 
+$(CRIU_LIB)/libcriu.so:
+	$(MAKE) -C ../../../../criu install-lib \
+		DESTDIR=$(INSTALL_DIR) PREFIX=/ LIBDIR=/lib \
+		SKIP_PIP_INSTALL=1
+
+$(CRIU_INCLUDE): $(CRIU_LIB)/libcriu.so
+	@:
+
 define genb
 $(1): $(1).o lib.o
-	gcc $$^ -L ../../../../criu/lib/c/ -L ../../../../criu/images/ -lcriu -o $$@
+	gcc $$^ $(LDFLAGS) -o $$@
 endef
 
 $(foreach t, $(TESTS), $(eval $(call genb, $(t))))
 
-%.o: %.c
-	gcc -c $^ -iquote ../../../../criu/criu/include -I../../../../criu/lib/c/ -I../../../../criu/images/ -o $@ -Werror
+%.o: %.c | $(CRIU_INCLUDE)
+	gcc -c $< $(CFLAGS) -o $@ -Werror
 
-clean: libcriu_clean
-	rm -rf $(TESTS) $(TESTS:%=%.o) lib.o
+clean:
+	rm -rf $(TESTS) $(TESTS:%=%.o) lib.o $(INSTALL_DIR)
 .PHONY: clean
-
-libcriu_clean:
-	rm -f libcriu.so.${CRIU_SO_VERSION_MAJOR}
-.PHONY: libcriu_clean
-
-libcriu:
-	ln -s ../../../../criu/lib/c/libcriu.so libcriu.so.${CRIU_SO_VERSION_MAJOR}
-.PHONY: libcriu

--- a/test/others/libcriu/run.sh
+++ b/test/others/libcriu/run.sh
@@ -14,12 +14,11 @@ source "${MAIN_DIR}/../env.sh" || exit 1
 
 echo "== Clean"
 make clean
-make libcriu
 
 rm -rf "${OUTPUT_DIR}"
 
 echo "== Run tests"
-export LD_LIBRARY_PATH=.
+export LD_LIBRARY_PATH=${MAIN_DIR}/.install/lib
 export PATH="${MAIN_DIR}/../../../criu:${PATH}"
 
 RESULT=0
@@ -76,6 +75,5 @@ fi
 run_test test_feature_check
 
 echo "== Tests done"
-make libcriu_clean
 [ "${RESULT}" -eq 0 ] && echo "Success" || echo "FAIL"
 exit "${RESULT}"

